CostManagement RP- Introducing API Version 2019-10-01 plus adding Modern Commerce Account Scopes to Query and Dimension APIs.

parisa-naeimi opened the pull request:
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/parisa-naeimi/azure-rest-api-specs/dev-cost-management-Microsoft.CostManagement-2019-10-01...master/).

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

parisa-naeimi committed:
Adds base for updating Microsoft.CostManagement from version stable/2019-09-01 to version 2019-10-01

parisa-naeimi committed:
Updates readme

parisa-naeimi committed:
Updates API version in new specs and examples

parisa-naeimi committed:
Updated Query and Dimension API to support GTM Scopes plus examples. Fixed an existing issue in Query API by removing value

parisa-naeimi committed:
Fixed a small issue

AutorestCI commented:
# Automation for azure-sdk-for-python

parisa-naeimi committed:
Fixed typo

parisa-naeimi committed:
Removing unused example

azuresdkci commented:
Can one of the admins verify this patch?

parisa-naeimi committed:
removed extra line

ArcturusZhang commented:
Hi @zikalino do you have any clue what is happening on the python SDK check?

ArcturusZhang commented:
Hi @parisa-naeimi please add a new entry in `readme.go.md` for this version of SDK.
Because of lack of new tag, the CI check seems to use mock information and crashes. I suppose adding a new entry in `readme.go.md` would fix this

ArcturusZhang reviewed (changes requested):
I add some comments, please have a look

parisa-naeimi committed:
Added new item for Go Package

AutorestCI commented:
# Automation for azure-sdk-for-go

parisa-naeimi committed:
removed nextlink

parisa-naeimi committed:
Fixed some minor issues

parisa-naeimi committed:
Fixed some minor issues and also added missing description for properties

parisa-naeimi commented:
> Hi @parisa-naeimi please add a new entry in `readme.go.md` for this version of SDK.
> Because of lack of new tag, the CI check seems to use mock information and crashes. I suppose adding a new entry in `readme.go.md` would fix this

@ArcturusZhang I already addressed your comment.

ArcturusZhang commented on specification/cost-management/resource-manager/readme.go.md:
````diff
@@ -49,3 +50,11 @@ Please also specify `--go-sdk-folder=<path to the root directory of your azure-s
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag) == 'package-2019-10' && $(go)
+output-folder: $(go-sdk-folder)/services/preview/$(namespace)/mgmt/2019-10-01/$(namespace)
````
Please change this to 
`$(go-sdk-folder)/services/preview/$(namespace)/mgmt/2019-10-01-preview/$(namespace)`

ArcturusZhang commented on specification/cost-management/resource-manager/readme.go.md:
````diff
@@ -49,3 +50,11 @@ Please also specify `--go-sdk-folder=<path to the root directory of your azure-s
 ``` yaml $(tag) == 'package-preview-2019-03' && $(go)
 output-folder: $(go-sdk-folder)/services/preview/$(namespace)/mgmt/2019-03-01/$(namespace)
````
Technically this should be changed too, but since this package has already been released, we just leave this here unchanged.

ArcturusZhang reviewed (changes requested):
I add some more comments, please have a look.
In the mean time, please wait for ARM team to review this new RP.

parisa-naeimi committed:
addressed some minor issues

parisa-naeimi committed:
added legacy vs modern

parisa-naeimi committed:
fixed a small issue

KrisBash reviewed (approved):
LGTM

ArcturusZhang commented:
/azp run
